testing/e2e/suites/defaultdomain.go

```diff
@@ -240,6 +242,76 @@ func defaultDomainTests(in infra.Provisioned) []test {
 					return fmt.Errorf("parsing and verifying TLS certificate: %w", err)
 				}
 
+				lgr.Info("Validating Rotation Story")
+				newCert, newKey, err := manifests.GenerateSelfSignedCert()
+				if err != nil {
+					return fmt.Errorf("generating self-signed cert: %w", err)
+				}
+
+				dds := manifests.CreateDefaultDomainSecret(newCert, newKey)
+				if err := util.Upsert(ctx, cl, dds); err != nil {
+					return fmt.Errorf("upserting DefaultDomainSecret: %w", err)
+				}
+
+				// Retry waiting for certificate rotation with timeout
+				if err := wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+					rotatedSecret := &corev1.Secret{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      secretTarget,
+							Namespace: namespace.GetName(),
+						},
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "Secret",
+							APIVersion: corev1.SchemeGroupVersion.String(),
+						},
+					}
+					if err := cl.Get(ctx, client.ObjectKeyFromObject(rotatedSecret), rotatedSecret); err != nil {
+						lgr.Info("failed to get rotated secret, retrying", "error", err.Error())
+						return false, nil // Retry on error
+					}
+
+					rotatedCert, ok := rotatedSecret.Data["tls.crt"]
+					if !ok {
+						lgr.Info("rotated secret does not contain tls.crt data, retrying")
+						return false, nil // Retry
+					}
+					rotatedKey, ok := rotatedSecret.Data["tls.key"]
+					if !ok {
+						lgr.Info("rotated secret does not contain tls.key data, retrying")
+						return false, nil // Retry
+					}
+
+					// Check if certificate matches the new one
+					if !bytes.Equal(rotatedCert, newCert) {
+						lgr.Info("rotated certificate does not match the upserted certificate yet, retrying")
+						return false, nil // Retry
+					}
+					if bytes.Equal(rotatedCert, tlsCert) {
+						lgr.Info("rotated certificate is still the same as the original, retrying")
+						return false, nil // Retry
+					}
+
+					// Check if key matches the new one
+					if !bytes.Equal(rotatedKey, newKey) {
+						lgr.Info("rotated key does not match the upserted key yet, retrying")
+						return false, nil // Retry
+					}
+					if bytes.Equal(rotatedKey, tlsKey) {
+						lgr.Info("rotated key is still the same as the original, retrying")
+						return false, nil // Retry
+					}
+
+					lgr.Info("Validating Rotated Certificate and Key")
+					if _, err := tls.ParseTLSCertificate(rotatedCert, rotatedKey); err != nil {
+						lgr.Info("failed to parse rotated TLS certificate, retrying", "error", err.Error())
```

Comment:
this would be really bad, let's error here. we shouldn't go from valid to invalid